### ISSUE TITLE
Long Windows paths can cause build failures

### ISSUE THREAD

**Issue by miguelriverarios** [google/shaka-player#3369]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
3.1.0-pre

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `master`?**
Yes

**Are you using the demo app or your own custom app?**
N/A

**If custom app, can you reproduce the issue using our demo app?**
N/A

**What browser and OS are you using?**
Google Chrome on Windows 10

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
N/A

**What are the manifest and license server URIs?**

**What configuration are you using?  What is the output of `player.getConfiguration()`?**
N/A

**What did you do?**
Following the instructions [here](https://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html)
1) git clone https://github.com/google/shaka-player.git
2) cd shaka-player
3) python build/all.py

**What did you expect to happen?**
The library to compile

**What actually happened?**
When I run python build/all.py I get the following error message:

```
Traceback (most recent call last):
  File "build/all.py", line 150, in <module>
    shakaBuildHelpers.run_main(main)
, in main
    if not step(parsed_args):
  File "C:\Users\Miguel Rivera Rios\Documents\Jobs\Sky\shaka-player\build\check.py", line 88, in check_js_lint
    return linter.lint(fix=args.fix, force=args.force)
  File "C:\Users\Miguel Rivera Rios\Documents\Jobs\Sky\shaka-player\build\compiler.py", line 331, in lint
    if shakaBuildHelpers.execute_get_code(cmd_line) != 0:
  File "C:\Users\Miguel Rivera Rios\Documents\Jobs\Sky\shaka-player\build\shakaBuildHelpers.py", line 173, in execute_get_code
    obj = execute_subprocess(args)
  File "C:\Users\Miguel Rivera Rios\Documents\Jobs\Sky\shaka-player\build\shakaBuildHelpers.py", line 162, in execute_subprocess
    return subprocess.Popen(args, **kwargs)
  File "C:\Python27\lib\subprocess.py", line 394, in __init__
    errread, errwrite)
  File "C:\Python27\lib\subprocess.py", line 644, in _execute_child
    startupinfo)
WindowsError: [Error 87] The parameter is incorrect
```

It specifies this as a WindowsError, but I'm not sure how to resolve.

**Comment by TheModMaker** [comment]
Can you try with Python 3?  How are you building it?  Are you using cmd, PowerShell, WSL, or CYGWIN?

**Comment by surajkumar-sk** [comment]
@miguelriverarios This is an issue with python in windows and has nothing to do with shaka-player. In python, the working directory is limited to MAX_PATH - 2 (258) characters for some reason I don't know. for reading more about this issue please follow this [link ](https://bugs.python.org/issue35678). A workaround for fixing this issue is having a short directory name. Clone the project in a folder with a shorter directory name something like "E:\sky\shaka-player" or "C:\Users\Miguel Rivera Rios\Documents\shaka-player". 

I might be wrong but it doesn't hurt to try.
try the above solution and please let me know the results.
Thank you

**Comment by joeyparrish** [comment]
@surajkumar-sk, thanks for trying to help out on this one, but it is clear from the console output in the original post that the directory is well within the MAX_PATH limit.  It looks to be 60 characters long.

@miguelriverarios, I'm unable to reproduce your results.  I'm using a cygwin bash shell and native Windows version of Python.  I tried both Python 3.8.0 and Python 2.7.11, and both worked.  Can you please answer the questions posed by @TheModMaker?

> Can you try with Python 3? Are you using cmd, PowerShell, WSL, or CYGWIN?

**Comment by surajkumar-sk** [comment]
@joeyparrish I am confused with this too. The issue on python's page say the max_limit is 258 but when I tried testing it on my windows the maximum path length allowed was only 47(including '/shaka-player'). Any path name greater than 47 throws the same error. I taught of investigating this futher if the author of this issue responds saying this fix worked. 
From the logs , the path length is greater than 42 so I think reducing the path length will work.

I can be wrong but I was able to reproduce this issue and reducing path length to less than 42 worked.

**Comment by joeyparrish** [comment]
@surajkumar-sk, I set out to prove you wrong.  I copied my shaka-player source folder on Windows to a folder called `very-long-pathname-indeed-to-prove-that-this-does-not-matter-at-all`.  My absolute path length was then 96.  (It is normally only 41.)

What I found surprised me:

```sh
$ python3 build/all.py --force
[INFO] Generating Closure dependencies...
[INFO] Linting JavaScript...
[ERROR] *** A required dependency is missing: node
```

It's not the same exact failure that you and @miguelriverarios experienced, but it seems related to the path length.  Also, I'm using Cygwin shell, native Windows Python 3.8.0, and native Windows node v12.18.3.  So perhaps a different environment means different output.

Let's let @miguelriverarios have a chance to reply with details on:

1. Their shell environment (cygwin, cmd, powershell, etc)
2. Their Python and node versions
3. Whether or not reducing the absolute path length has an effect on the issue

Thanks!
